cmake/modules/AddSwift.cmake

```diff
@@ -516,6 +529,8 @@ function(_add_swift_lipo_target)
     ""                  # multi-value args
     ${ARGN})
 
+  is_darwin_based_sdk("${LIPO_SDK}" IS_DARWIN)
+  precondition(IS_DARWIN MESSAGE "It's not possible to build a universal binary for a non-MachO target")
```

Comment:
Can you get away with:

```
precondition(${LIPO_SDK} IN_LIST SWIFT_APPLE_PLATFORMS MESSAGE ...)
```

Reply:
Seems that the simple solution doesn’t work, because `precondition` is an user-defined function, so the condition is broken up in pieces. Probably I will have to save the value into a variable before testing.